src/target/stm32f1.c

```diff
@@ -254,6 +257,136 @@ bool at32fxx_probe(target_s *t)
 	return false;
 }
 
+/* mm32l0 flash write */
+
+void mm32l0_mem_write_sized(adiv5_access_port_s *ap, uint32_t dest, const void *src, size_t len, align_e align)
+{
+	uint32_t odest = dest;
+
+	len >>= align;
+	ap_mem_access_setup(ap, dest, align);
+	while (len--) {
+		uint32_t tmp = 0;
+		/* Pack data into correct data lane */
+		switch (align) {
+		case ALIGN_BYTE: {
+			uint8_t value;
+			memcpy(&value, src, sizeof(value));
+			tmp = (uint32_t)value << (8U * (dest & 3U));
+			break;
+		}
+		case ALIGN_HALFWORD: {
+			uint16_t value;
+			memcpy(&value, src, sizeof(value));
+			tmp = (uint32_t)value; /* MM32 special, no shift << (8U * (dest & 2U)) */
+			break;
+		}
+		case ALIGN_DWORD:
+		case ALIGN_WORD:
+			memcpy(&tmp, src, sizeof(tmp));
+			break;
+		}
+		src = (uint8_t *)src + (1 << align);
+		dest += (1 << align);
+		adiv5_dp_low_access(ap->dp, ADIV5_LOW_WRITE, ADIV5_AP_DRW, tmp);
+
+		/* Check for 10 bit address overflow */
+		if ((dest ^ odest) & 0xfffffc00U) {
+			odest = dest;
+			adiv5_dp_low_access(ap->dp, ADIV5_LOW_WRITE, ADIV5_AP_TAR, dest);
+		}
+	}
+	/* Make sure this write is complete by doing a dummy read */
+	adiv5_dp_read(ap->dp, ADIV5_DP_RDBUFF);
+}
+
+/* Identify MM32 devices (Cortex-M0) */
+
+bool mm32l0xx_probe(target_s *t)
+{
+	uint32_t mm32_id;
+	const char *name = "?";
+	size_t flash_kbyte = 0;
+	size_t ram_kbyte = 0;
+	size_t block_size = 0x400U;
+
+	if (target_mem_read(t, &mm32_id, DBGMCU_IDCODE_MM32L0, sizeof(mm32_id))) {
```

Comment:
Just so we understand it better, what's the motivation of using target_mem_read() here instead of target_mem_read32() + target_check_error()?